cub/cub/block/block_scan.cuh

```diff
@@ -1291,177 +1549,1368 @@ public:
     detail::ThreadScanExclusive(input, output, scan_op, thread_prefix);
   }
 
-  //! @}  end member group
```

Comment:
The diff here on GH is messed up again, should be almost purely positive. See in VS Code "GitHub Pull Request" extension.

Reply:
FYI: Most of the ~2500 lines added in this file (and therefore probably almost half of the ~5000 total lines added in this PR) are docstrings including code examples which were adapted from the existing APIs.

I know that we would prefer to have all our examples tested but this would have multiplied the amount of work for this PR given that the given examples can't just be tested in their current "incomplete" form (e.g. they are not specifying how data is loaded or stored).